src/cmd/update.rs

```diff
@@ -0,0 +1,68 @@
+use std::fs;
+use std::fs::File;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+
+use anyhow::Result;
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const RELEASES_LIST: &str = "https://api.github.com/repos/sbstp/kubie/contents/releases/linux?ref=master";
+const RELEASE_BASE_URL: &str = "https://github.com/sbstp/kubie/raw/master/releases/linux/amd64";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+struct TreeUrl {
+    git_url: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Tree {
+    tree: Vec<KubieVersion>,
+}
+
+#[derive(Debug, Deserialize)]
+struct KubieVersion {
+    path: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_version = get_latest_version()?;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let resp = attohttpc::get(format!("{}/{}/{}", RELEASE_BASE_URL, latest_version, FILENAME)).send()?;
+        if resp.is_success() {
+            let file = File::create(FILENAME)?;
+            let _n = resp.write_to(file)?;
```

Comment:
If you don't need the amount of bytes written you can remove the `let` binding, but keep the error handling. I'm guessing this might fail if kubie is not called with sudo? Perhaps you can add `context` to this error as well.

Reply:
Indeed I don't need it anymore :) Removed the binding.